docs/modules/servers/pages/postgres/objectives.adoc

```diff
@@ -0,0 +1,22 @@
+= Distributed James Server &mdash; Objectives and motivation
+:navtitle: Objectives and motivation
+
+From the outstanding advantages of a distributed mail system, such as scalability and enhancement,
+this project aims to implement a backend database version using Postgres,
+
+Primary Objectives:
+
+* Provide more options: The current James Distributed server uses Cassandra as the backend database.
+  This project aims to provide an alternative to Cassandra, using Postgres as the backend database.
+  The choice of version depends on the operator's decision.
+* Replace the `jpa-app` variant: The `jpa-app` variant is a simple version of James that uses JPA to store data.
+  This variant has not been updated, maintained, or tested for a long time.
+  With postgres-app, we use the `r2dbc` library to connect to the Postgres database, which implement non-blocking, reactive APIs.
+  It is higher performance.
+* Leverage advanced Postgres features: Postgres is a powerful database that supports many advanced features.
+  This project aims to leverage these features to improve the efficiency of the James server.
+  For example, the implement https://www.postgresql.org/docs/current/ddl-rowsecurity.html[row-level security]
+  to improve the security of the James server.
+* Flexible deployment: The new architecture allows flexible module choices. You can use Postgres directly for
+  blob storage or use Object Storage (e.g Minio, S3...).
+  You can use Postgres's search query features directly or opt for OpenSearch.
```

Comment:
Q: What does `Postgres's search query features` mean?

Here I would understand we implemented the whole searching using Postgres.

Reply:
> Q: What does Postgres's search query features mean?

It just a SQL Queries 
Where ... like... =, equal ...

Reply:
Hm except that actually currently we don't have search via postgresql. Opensearch or memory so far?

Could be a topic though that we could groom

Reply:
Isn't this right?
```
org.apache.james.mailbox.postgres.search.AllSearchOverride,org.apache.james.mailbox.postgres.search.DeletedSearchOverride, org.apache.james.mailbox.postgres.search.DeletedWithRangeSearchOverride,org.apache.james.mailbox.postgres.search.NotDeletedWithRangeSearchOverride,org.apache.james.mailbox.postgres.search.UidSearchOverride,org.apache.james.mailbox.postgres.search.UnseenSearchOverride
```

Reply:
Hmm, that is just a part of the search index. This makes me understand that one can use Postgres fully for search index.

Reply:
It's not really a search, it's overrides with proper tables to find particular info faster via the db instead of doing a normal search.

There is no proper usage of the search function in postgres though yet

Reply:
removed it!